tests/src/test/scala/org/apache/spark/sql/rapids/TestTrampolineUtil.scala

```diff
@@ -16,13 +16,21 @@
 
 package org.apache.spark.sql.rapids
 
-import org.apache.spark.sql.{DataFrame, Dataset, SparkSession}
+import org.apache.spark.sql.DataFrame
 import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
+import org.apache.spark.sql.rapids.shims.TrampolineConnectShims
 
 object TestTrampolineUtil {
-  def toLogicalPlan(df: DataFrame): LogicalPlan = df.logicalPlan
+  def toLogicalPlan(df: DataFrame): LogicalPlan = {
+    // Use the shim method and handle type casting internally
+    val shimDf = df.asInstanceOf[TrampolineConnectShims.DataFrame]
+    TrampolineConnectShims.getLogicalPlan(shimDf)
+  }
 
-  def toDataFrame(spark: SparkSession, plan: LogicalPlan): DataFrame = {
-    Dataset.ofRows(spark, plan)
+  def toDataFrame(spark: Any, plan: LogicalPlan): DataFrame = {
+    // Use the shim method and handle type casting internally
+    TrampolineConnectShims.createDataFrame(
+      spark.asInstanceOf[TrampolineConnectShims.SparkSession],
+      plan).asInstanceOf[DataFrame]
```

Comment:
@gerashegalov - Since this in tests module, I am casting it directly to DataFrame (as it can be considered as safe).
Please let me know if this may cause any issues and if there are the other options to solve.

Reply:
Thanks @gerashegalov for pairing over this problem. I have updated the PR with the suggested fixes. PTAL.